src/overlays/actors/ovl_Dm_Opstage/z_dm_opstage.c

```diff
@@ -32,26 +31,108 @@ const ActorInit Dm_Opstage_InitVars = {
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Dm_Opstage/func_80A9FA58.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Dm_Opstage/DmOpstage_Update.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Dm_Opstage/DmOpstage_Draw.s")
+void DmOpstage_SetupAction(DmOpstage* this, DmOpstageActionFunc actionFunc) {
+    this->actionFunc = actionFunc;
+}
+
+void DmOpstage_Init(Actor* thisx, GlobalContext* globalCtx) {
+    s32 params;
+    DmOpstage* this = THIS;
+
+    Actor_ProcessInitChain(&this->dyna.actor, sInitChain);
+    DmOpstage_SetupAction(this, func_80A9FA58);
+    Actor_SetScale(&this->dyna.actor, 0.1f);
+    params = DMOPSTAGE_GET_FF;
+    if (params == 0) {
+        DynaPolyActor_Init(&this->dyna, 0);
+        DynaPolyActor_LoadMesh(globalCtx, &this->dyna, &object_keikoku_demo_Colheader_001C98);
+        params = DMOPSTAGE_GET_FF;
+    }
+    if (params > 0) {
+        this->posX = this->dyna.actor.world.pos.x;
+        this->posY = this->dyna.actor.world.pos.y;
+        this->posZ = this->dyna.actor.world.pos.z;
+        this->dyna.actor.world.pos.x = 0.0f;
+        this->dyna.actor.world.pos.y = 0.0f;
+        this->dyna.actor.world.pos.z = 0.0f;
+    }
+}
+
+void DmOpstage_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+    DmOpstage* this = THIS;
+
+    if (DMOPSTAGE_GET_FF == 0) {
+        DynaPoly_DeleteBgActor(globalCtx, &globalCtx->colCtx.dyna, this->dyna.bgId);
+    }
+}
+
+void func_80A9FA58(DmOpstage* this, GlobalContext* globalCtx) {
+    s16 params;
+    s32 actionIndex;
+
+    params = this->dyna.actor.params;
+    if ((params & 0xFF) == 0) {
+        if (Cutscene_CheckActorAction(globalCtx, 115)) {
+            actionIndex = Cutscene_GetActorActionIndex(globalCtx, 115);
+            if (globalCtx->csCtx.actorActions[actionIndex]->action == 2) {
+                this->dyna.actor.scale.x = 0.075f;
+                this->dyna.actor.scale.z = 0.3f;
+            } else {
+                this->dyna.actor.scale.x = 0.1f;
+                this->dyna.actor.scale.z = 0.1f;
+            }
+            Cutscene_ActorTranslateAndYaw(&this->dyna.actor, globalCtx, actionIndex);
+        }
+    } else if (Cutscene_CheckActorAction(globalCtx, (((params >> 8) & 0xFF) + 0x74) & 0xFFFF)) {
+        Cutscene_ActorTranslateAndYaw(
+            &this->dyna.actor, globalCtx,
+            Cutscene_GetActorActionIndex(globalCtx, (((this->dyna.actor.params >> 8) & 0xFF) + 0x74) & 0xFFFF));
+    }
+}
+
+void DmOpstage_Update(Actor* thisx, GlobalContext* globalCtx) {
+    DmOpstage* this = THIS;
+
+    this->actionFunc(this, globalCtx);
+    if ((globalCtx->sceneNum == 8) && (gSaveContext.sceneSetupIndex == 0) && (globalCtx->csCtx.frames == 480)) {
+        func_8019F128(NA_SE_EV_NAVY_FLY_REBIRTH);
+    }
+}
+
+void DmOpstage_Draw(Actor* thisx, GlobalContext* globalCtx) {
+    s32 params;
+    DmOpstage* this = THIS;
+    params = DMOPSTAGE_GET_FF;
```

Comment:
```suggestion
    DmOpstage* this = THIS;
    s32 params = DMOPSTAGE_GET_FF(thisx);
```
Does this match? You can typically move variable assignments up to their declare if there are no non-assigment statements between them.

Except we try to keep this = THIS at the top when possible.

Reply:
![image](https://user-images.githubusercontent.com/96613413/172864142-909b4c82-d15c-4c2c-8152-86af1e130db6.png)
it only works with (this), thisx macro doesn't compile in any part of the code.

Reply:
thisx will compile when you correct the `DMOPSTAGE_GET_TYPE` macro

Reply:
okay, fixed the macro, still doesn't match with s32 params = DMOPSTAGE_GET_TYPE(thisx);
so i'll keep it the way it was but this time with the corrected macro.